app/iOS/Nouns/Modules/Components/Madhappy/MadhappyBanner.swift

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2022 Nouns Collective
+//
+// Originally authored by Mohammed Ibrahim
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import SwiftUI
+import NounsUI
+import SpriteKit
+
+struct MadhappyBanner: View {
+  
+  @Binding var isBottomSheetPresented: Bool
+  
+  static func shouldShowBanner(currentDate: Date) -> Bool {    
+    var calendar = Calendar(identifier: .gregorian)
+    calendar.timeZone = .init(identifier: "America/Los_Angeles") ?? .current
+            
+    // November 28, 2022 - 9:00 AM PDT
+    let startDateComponents = DateComponents(year: 2022, month: 11, day: 28, hour: 9, minute: 0, second: 0)
+    
+    // November 29, 2022 - 9:00 AM PDT
+    let endDateComponents = DateComponents(year: 2022, month: 11, day: 29, hour: 9, minute: 0, second: 0)
+    
+    guard let startDate = calendar.date(from: startDateComponents),
+          let endDate = calendar.date(from: endDateComponents) else {
+      return false
+    }
+    
+    return (startDate...endDate).contains(currentDate)
+  }
+    
+  var body: some View {
+    TimelineView(.periodic(from: .now, by: 2)) { context in
```

Comment:
The `TimelineView` itself is only rendered once, so I tried putting a random time function in place of the `2` value here but it only gets called once. SKSpriteNodes handle it a bit differently I suppose - not impossible here, but it would require replacing TimelineView with something else and I'm not sure what that would be.

Reply:
No worries, not a huge deal.